artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -2752,7 +2753,27 @@ public String listAllConsumersAsJSON() throws Exception {
    }
 
    private JsonObject toJSONObject(ServerConsumer consumer) throws Exception {
-      JsonObjectBuilder obj = JsonLoader.createObjectBuilder().add("consumerID", consumer.getID()).add("connectionID", consumer.getConnectionID().toString()).add("sessionID", consumer.getSessionID()).add("queueName", consumer.getQueue().getName().toString()).add("browseOnly", consumer.isBrowseOnly()).add("creationTime", consumer.getCreationTime()).add("deliveringCount", consumer.getDeliveringMessages().size());
+      List<MessageReference> deliveringMessages = consumer.getDeliveringMessages();
```

Comment:
this is going to hold a lock on the consumer while the messages are copied over.


if you have a bad consumer leaking on acks, this could make things worse to the broker.

Besides this will hold a lock on the ServerConsumer.

Perhaps I'm being over zealous here... but I have seen cases where using the console to find what was wrong actually made things worse.

That's my only concern here.

Reply:
it would be better if these metrics were pre-calculated in the Consumer and the console would just get them.